elasticdl/python/master/task_dispatcher.py

```diff
@@ -346,12 +354,20 @@ def report(self, request, success):
             if success:
                 if task in self._task_retry_count:
                     del self._task_retry_count[task]
-                if self._callbacks_list.model.stop_training:
-                    # Clear todo list to stop training
-                    self._todo = []
 
         return (time.time() - start_time), task, worker_id
 
+    def _check_exceed_max_step(self, task):
+        if self._max_step <= 0:
```

Comment:
`if self._max_step <=  0 or task.type != elasticdl_pb2.TRAINING:`

so we can omit the second if later.

Reply:
`if self._max_step > 0 and task.type == elasticdl_pb2.TRAINING`